src/msg/async/AsyncConnection.cc

```diff
@@ -151,6 +151,11 @@ AsyncConnection::~AsyncConnection()
   if (recv_buf)
     delete[] recv_buf;
   ceph_assert(!delay_state);
+  // protocol must be deallocated in proper EventCenter. This is ensured
```

Comment:
what about `recv_buf`?

Reply:
Apparently, it isn't touched outside the EC. Constructor only allocates the memory.